File: src/ThePit.DataAccess/Entities/Invoice.cs
namespace ThePit.DataAccess.Entities;

public class Invoice
{
    public int Id { get; set; }
    public string InvoiceNumber { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? PaidAt { get; set; }
    public string Status { get; set; } = "Pending";
    public string CustomerName { get; set; } = string.Empty;
    public string CustomerEmail { get; set; } = string.Empty;
}

File: src/thepit-ui/src/lib/types/invoice.ts
export interface Invoice {
	id: number;
	invoiceNumber: string;
	amount: number;
	dueDate: string;
	status: InvoiceStatus;
	createdAt: string;
}

export type InvoiceStatus = 'Pending' | 'Paid' | 'Overdue' | 'Cancelled';

export interface CreateInvoiceDto {
	invoiceNumber: string;
	amount: number;
	dueDate: string;
}

export interface UpdateInvoiceDto {
	invoiceNumber?: string;
	amount?: number;
	dueDate?: string;
	status?: InvoiceStatus;
}
